### ISSUE TITLE
Include `endpoints.enabled` checks in autoconfigure classes

### ISSUE THREAD

**Issue by jonasholtkamp** [spring-cloud/spring-cloud-commons#264]
Hi guys,

the classes in `o.s.cloud.autoconfigure` (e.g. `LifecycleMvcEndpointAutoConfiguration`) manage the creation of Spring Cloud's [management endpoints](http://cloud.spring.io/spring-cloud-static/Dalston.SR4/multi/multi__spring_cloud_context_application_context_services.html#_endpoints). However, unlike Spring Boot Actuator's classes in `o.s.boot.actuate.autoconfigure` Cloud's classes don't follow the property `endpoints.enabled`. This leads to inconsistent behaviour.

I disable all Actuator endpoints (e.g. to not expose them in production stage) like this:
```
endpoints:
  enabled: false
```
This works well, all Actuator-provided endpoints respond with a 404. Yet I still have to disable all of Cloud's endpoints by themselves, otherwise they're still available, which is a bit cumbersome:
```
endpoints:
  refresh.enabled: false
  restart.enabled: false
  pause.enabled: false
  resume.enabled: false
  env.post.enabled: false
```

Apparently there is an [Github issue](https://github.com/spring-projects/spring-boot/issues/6383) in the Spring Boot repository exactly about this issue which has been closed with the rationale of being out of Boot's control.

My proposal would be, that Cloud's above mentioned classes stick to the property `endpoints.enabled` like Boot does with the condition class `o.s.boot.actuate.condition.OnEnabledEndpointCondition`.

Additionally (maybe another issue): I'd prefer that upon deactivation not only the POST endpoints would respond with an HTTP 404 but also all other HTTP verbs like GET. Current behaviour (whether activated or deactivated) is an HTTP 405 response upon calling e.g. `GET /restart`. Preferred behaviour when deactivated: `GET /restart` -> HTTP 404.

Regards

**Comment by spencergibb** [comment]
We have tests that show we disable these endpoints globally. https://github.com/spring-cloud/spring-cloud-commons/blob/master/spring-cloud-context/src/test/java/org/springframework/cloud/autoconfigure/LifecycleMvcAutoConfigurationTests.java

What version are you using?

**Comment by PeterWippermann** [comment]
On behalf of @jonasholtkamp I can say that we're on version `1.1.8`.

So that feature was added with #208, I guess? And thus its available from `1.2.2`?

**Comment by jonasholtkamp** [comment]
I can confirm that. We are using 1.1.8 (from Camden.SR6). When using 1.2.2 (from Dalston.SR4) the endpoints are disabled too when `endpoints.enabled` is set to `false`.

Thanks for your help.
